server/web/router.go

```diff
@@ -455,32 +455,35 @@ func (p *ControllerRegister) RouterAny(pattern string, f interface{}) {
 //    }
 //
 //    AddRouterMethod("get","/api/:id", MyController.Ping)
-func (p *ControllerRegister) AddRouterMethod(method, pattern string, f interface{}) {
-	method = strings.ToUpper(method)
-	if method != "*" && !HTTPMETHOD[method] {
-		panic("not support http method: " + method)
+func (p *ControllerRegister) AddRouterMethod(httpMethod, pattern string, f interface{}) {
+	httpMethod = strings.ToUpper(httpMethod)
+	if httpMethod != "*" && !HTTPMETHOD[httpMethod] {
+		panic("not support http method: " + httpMethod)
 	}
 
 	ct, methodName := getReflectTypeAndMethod(f)
+	p.addBeegoTypeRouter(ct, methodName, httpMethod, pattern)
+}
+
+// addBeegoTypeRouter add beego type router
+func (p *ControllerRegister) addBeegoTypeRouter(ct reflect.Type, ctMethod, httpMethod, pattern string) {
 	route := &ControllerInfo{}
 	route.pattern = pattern
 	route.routerType = routerTypeBeego
 	route.sessionOn = p.cfg.WebConfig.Session.SessionOn
 	route.controllerType = ct
 
 	methods := make(map[string]string)
```

Comment:
could you refactor this part because I found that there are many similar code:
![image](https://user-images.githubusercontent.com/9923838/104202429-e37d4580-5465-11eb-8903-ef72a27d7bbf.png)
![image](https://user-images.githubusercontent.com/9923838/104202442-e841f980-5465-11eb-9edf-dd3f563fdce0.png)
![image](https://user-images.githubusercontent.com/9923838/104202479-f263f800-5465-11eb-94f3-d181315516b0.png)